### PR TITLE
NH-63239: Add optional opentelemetry-ebpf to SWI K8s collector

### PR DESCRIPTION
**Pull request by gantrior**
this is based on https://github.com/solarwinds/swi-k8s-opentelemetry-collector/pull/424 and rely on image update https://github.com/solarwinds/swi-k8s-opentelemetry-collector/pull/423

* Adding eBPF resources:
    * kernel collector (added to node collector daemonset as separate container) - it collects actual network usage and send data to reducer
    * k8s collector deployment - send to reduced information about k8s workloads
    * reducer - it merges data from k8s collector and kernel collector and publish data as OTLP metrics. Send them to metrics collector
* Adding appropriate `values.yaml` update
* Adding Changelog and documentation of emitted metrics (they are in raw form as they are emitted from [opentelemetry-ebpf](https://github.com/open-telemetry/opentelemetry-ebpf), but prefixed with `k8s.`)

### CODE REVIEW

File: deploy/helm/templates/node-collector-daemon-set.yaml [hunk 1]
```diff
@@ -129,7 +133,55 @@ spec:
               readOnly: true
             - name: logcheckpoints
               mountPath: {{ .Values.otel.logs.filestorage.directory }}
+{{- end }}
+{{- if and .Values.ebpfNetworkMonitoring.enabled .Values.ebpfNetworkMonitoring.kernelCollector.enabled }}
+        - name: swi-kernel-collector
+          image: "otel/opentelemetry-ebpf-kernel-collector:v0.10.0"
```
Comment (jaroslav-fedor-swi) [inline deploy/helm/templates/node-collector-daemon-set.yaml:139]: Should we make this a variable?

File: deploy/helm/templates/network/reducer-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            - --disable-metrics={{join "," .Values.ebpfNetworkMonitoring.reducer.disableMetrics}}
+            {{- end }}
+            {{- if .Values.ebpfNetworkMonitoring.reducer.enableMetrics }}
+            - --enable-metrics={{join "," .Values.ebpfNetworkMonitoring.reducer.enableMetrics}}
+            {{- end }}
+          env:
+            - name: "EBPF_NET_CRASH_METRIC_HOST"
+              value: "{{ include "common.fullname" (tuple . "-collector") }}"
+            - name: "EBPF_NET_CRASH_METRIC_PORT"
+              value: "4317"
```
Comment (jaroslav-fedor-swi) [inline deploy/helm/templates/network/reducer-deployment.yaml:53]: Should we make this port a variable?

File: deploy/helm/templates/node-collector-daemon-set.yaml (hunk 1 above)
Comment (gantrior) [reply deploy/helm/templates/node-collector-daemon-set.yaml:139]: I wasn't sure if there is use case for it, but yeah for better flexibility I will add option to `values.yaml` to be able to override images

File: deploy/helm/templates/collector-service.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "common.fullname" (tuple . "-collector") }}
```
[the remark this answers is not in the document]
Comment (gantrior) [reply deploy/helm/templates/collector-service.yaml:4]: Yes, that would be better name

[the remark this answers is not in the document]
Comment (gantrior) [reply deploy/helm/templates/collector-service.yaml]: fixed

File: deploy/helm/templates/node-collector-daemon-set.yaml (hunk 1 above)
Comment (gantrior) [reply deploy/helm/templates/node-collector-daemon-set.yaml]: fixed

File: deploy/helm/templates/network/reducer-deployment.yaml (hunk 2 above)
Comment (gantrior) [reply deploy/helm/templates/network/reducer-deployment.yaml]: fixed

File: deploy/helm/templates/metrics-deployment.yaml [hunk 4]
```diff
@@ -111,8 +111,11 @@ spec:
                   name: solarwinds-api-token
                   key: SOLARWINDS_API_TOKEN
                   optional: true
-{{- if .Values.otel.metrics.telemetry.metrics.enabled }}
           ports:
+            - name: otlp
+              containerPort: 4317
```
Comment (jaroslav-fedor-swi) [inline deploy/helm/templates/metrics-deployment.yaml:116]:
```suggestion
              containerPort: {{ .Values.otel.metrics.otlp_endpoint.port }}
```

File: deploy/helm/values.yaml [hunk 5]
```diff
@@ -437,3 +444,53 @@ aws_fargate:
     # see https://docs.fluentbit.io/manual/pipeline/filters
     # NOTE: The FluentBit configuration expects four spaces as indentation within sections
     filters:
+
+ebpfNetworkMonitoring:
+  enabled: false
```
Comment (jaroslav-fedor-swi) [inline deploy/helm/values.yaml:449]: We should enable it by default.

Comment (gantrior) [reply deploy/helm/values.yaml:449]: We need to discuss this offline, there are several risks involved in that

**Review by jaroslav-fedor-swi — DISMISSED** [review]

**Review by jaroslav-fedor-swi — APPROVED** [review]

**Pull request merged**